### DISCUSSION TITLE
Make StorageFile.GetFileFromPathAsync easier to use

### DISCUSSION THREAD

**Discussion by riverar** [microsoft/WindowsAppSDK#919] [Ideas]
Related to #8 

`StorageFile.GetFileFromPathAsync` is a bit painful to use. I'm thinking of the following quality of life changes, thoughts welcome:

* Remove absolute path restriction. (PR apps don't broker their file I/O, this restriction seems a bit silly.)
* Fix inability to accept paths with forward slashes. (Windows supports both `/` and `\` as a path separator, so it's not clear why this API is so rigid.)
* Fix inability to accept long paths (`\\?\`)

**Comment by jonwis** [comment]
@smaillet-ms - anything you'd like to add here? @riverar we don't have a Project Reunion based `StorageFile` implementation (yet!) but I'll file some internal issues to track improving the in-box one.

**Reply by riverar** [reply]
@jonwis Thanks. Some Reunion APIs, [like XamlBinaryWriter.Write](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.markup.xamlbinarywriter.write), require IRandomAccessStream, which makes things difficult. 

Quick search through Reunion winmds:
```
findstr /s /i /m Windows.Storage *

Microsoft.UI\Microsoft\UI\Xaml\Markup\IXamlBinaryWriterStatics
Microsoft.UI\Microsoft\UI\Xaml\Markup\XamlBinaryWriter
Microsoft.UI\Microsoft\UI\Xaml\Media\ILoadedImageSurfaceStatics
Microsoft.UI\Microsoft\UI\Xaml\Media\Imaging\BitmapSource
Microsoft.UI\Microsoft\UI\Xaml\Media\Imaging\IBitmapSource
Microsoft.UI\Microsoft\UI\Xaml\Media\Imaging\IRenderTargetBitmap
Microsoft.UI\Microsoft\UI\Xaml\Media\Imaging\ISvgImageSource
Microsoft.UI\Microsoft\UI\Xaml\Media\Imaging\IWriteableBitmap
Microsoft.UI\Microsoft\UI\Xaml\Media\Imaging\RenderTargetBitmap
Microsoft.UI\Microsoft\UI\Xaml\Media\Imaging\SvgImageSource
Microsoft.UI\Microsoft\UI\Xaml\Media\Imaging\WriteableBitmap
Microsoft.UI\Microsoft\UI\Xaml\Media\LoadedImageSurface
Microsoft.UI.Text\Microsoft\UI\Text\ITextDocument
Microsoft.UI.Text\Microsoft\UI\Text\ITextRange
Microsoft.UI.Text\Microsoft\UI\Text\RichEditTextDocument
Microsoft.UI.Text\Microsoft\UI\Text\RichEditTextRange
Microsoft.Web.WebView2.Core\Microsoft\Web\WebView2\Core\CoreWebView2
Microsoft.Web.WebView2.Core\Microsoft\Web\WebView2\Core\CoreWebView2Environment
Microsoft.Web.WebView2.Core\Microsoft\Web\WebView2\Core\CoreWebView2WebResourceRequest
Microsoft.Web.WebView2.Core\Microsoft\Web\WebView2\Core\CoreWebView2WebResourceResponse
Microsoft.Web.WebView2.Core\Microsoft\Web\WebView2\Core\CoreWebView2WebResourceResponseView
Microsoft.Web.WebView2.Core\Microsoft\Web\WebView2\Core\ICoreWebView2
Microsoft.Web.WebView2.Core\Microsoft\Web\WebView2\Core\ICoreWebView2Environment
Microsoft.Web.WebView2.Core\Microsoft\Web\WebView2\Core\ICoreWebView2WebResourceRequest
Microsoft.Web.WebView2.Core\Microsoft\Web\WebView2\Core\ICoreWebView2WebResourceResponse
Microsoft.Web.WebView2.Core\Microsoft\Web\WebView2\Core\ICoreWebView2WebResourceResponseView
```

**Reply by jonwis** [reply]
Check out https://docs.microsoft.com/en-us/windows/win32/api/shcore/nf-shcore-createrandomaccessstreamonfile and https://docs.microsoft.com/en-us/windows/win32/api/shcore/nf-shcore-createrandomaccessstreamoverstream which should get you an `IRandomAccessStream` over either a file path or an IStream of your choosing.

**Comment by ghost** [comment]
>  *  Fix inability to accept long paths (\\?\) 

good luck  with enforcing security at the subsystem level when  apps have direct access to NT UNC(read unsafe) paths.

**Reply by riverar** [reply]
Access checks can still be performed as usual. It's also irrelevant for full trust desktop apps, which are all Project Reunion apps at the moment.
